extensions/blocks/logo-gallery/edit.js

```diff
@@ -0,0 +1,306 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { filter, pick, map, get } from 'lodash';
+import {
+	IconButton,
+	PanelBody,
+	Toolbar,
+	withNotices,
+	DropZone,
+	FormFileUpload,
+} from '@wordpress/components';
+import {
+	BlockControls,
+	BlockIcon,
+	MediaPlaceholder,
+	MediaUpload,
+	mediaUpload,
+	InspectorControls,
+} from '@wordpress/editor';
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+import { isBlobURL } from '@wordpress/blob';
+import { Component, Fragment } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Logo from './logo';
+import icon from './icon';
+import './editor.scss';
+
+const ALLOWED_MEDIA_TYPES = [ 'image' ];
+
+export const pickRelevantMediaFiles = image => {
+	const imageProps = pick( image, [ 'alt', 'id', 'link' ] );
+	imageProps.url =
+		get( image, [ 'sizes', 'large', 'url' ] ) ||
+		get( image, [ 'media_details', 'sizes', 'large', 'source_url' ] ) ||
+		image.url;
+	return imageProps;
+};
+
+class GalleryEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onSelectImage = this.onSelectImage.bind( this );
+		this.onSelectImages = this.onSelectImages.bind( this );
+		this.onRemoveImage = this.onRemoveImage.bind( this );
+		this.setImageAttributes = this.setImageAttributes.bind( this );
+		this.setAttributes = this.setAttributes.bind( this );
+		this.addFiles = this.addFiles.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+
+		this.state = {
+			selectedImage: null,
+		};
+	}
+
+	setAttributes( attributes ) {
+		if ( attributes.ids ) {
```

Comment:
The background for needing these IDs is here: https://github.com/Automattic/jetpack/blob/28a898d10ac99bbbc7ad5342df0238198caaafe6/class.jetpack-post-images.php#L350-L356

Basically they're used to generate `og:image` tags. They might be useful for other purposes in future as well so even if you don't need them for this purpose, it might be nice to leave them in.

Reply:
As discussed, leaving the ids attribute in for potential future use and consistency between gallery type blocks.